one.jwr.interstellar.yaml

```diff
@@ -0,0 +1,132 @@
+      - python3 waf build
+      - python3 waf install
+    sources:
+      - type: git
+        url: https://github.com/mpv-player/mpv.git
+        tag: v0.35.1
+      - type: file
+        url: https://waf.io/waf-2.0.25
+        sha256: 21199cd220ccf60434133e1fd2ab8c8e5217c3799199c82722543970dc8e38d5
+        dest-filename: waf
+    modules:
+      - name: libass
+        cleanup:
+          - /include
+          - /lib/*.la
+          - /lib/pkgconfig
+        config-opts:
+          - --disable-static
+        sources:
+          - type: archive
+            url: https://github.com/libass/libass/releases/download/0.17.1/libass-0.17.1.tar.xz
+            sha256: f0da0bbfba476c16ae3e1cfd862256d30915911f7abaa1b16ce62ee653192784
+        modules:
+          - name: fribidi
+            cleanup:
+              - /bin
+              - /include
+              - /lib/pkgconfig
+              - /lib/*.la
+              - /share/man
+            buildsystem: meson
+            config-opts:
+              - --buildtype=release
+              - -Ddocs=false
+            sources:
+              - type: git
+                url: https://github.com/fribidi/fribidi.git
+                tag: v1.0.13
+                commit: b54871c339dabb7434718da3fed2fa63320997e5
+      - name: x264
+        cleanup:
+          - /include
+          - /lib/pkgconfig
+          - /share/man
+        config-opts:
+          - --disable-cli
+          - --enable-shared
+        sources:
+          - type: git
+            url: https://code.videolan.org/videolan/x264.git
+            commit: c1c9931dc87289b8aeba78150467f17bdb97d019
+      - name: nv-codec-headers
+        cleanup:
+          - '*'
+        no-autogen: true
+        make-install-args:
+          - PREFIX=/app
+        sources:
+          - type: git
+            url: https://github.com/FFmpeg/nv-codec-headers.git
+            commit: 855f8263d97bbdcaeabaaaa2997e1ccad7c52dc3
+      - name: ffmpeg
+        cleanup:
+          - /include
+          - /lib/pkgconfig
+          - /share/ffmpeg/examples
+        config-opts:
+          - --enable-shared
+          - --disable-static
+          - --enable-gnutls
+          - --disable-doc
+          - --disable-programs
+          - --disable-encoders
+          - --disable-muxers
+          - --enable-encoder=png
+          - --enable-libv4l2
+          - --enable-libdav1d
+        sources:
+          - type: git
+            url: https://git.ffmpeg.org/ffmpeg.git
+            tag: n6.0.1
+  - name: interstellar
+    buildsystem: simple
+    only-arches:
+      - x86_64
+    build-commands:
+      - 'cp -r interstellar/ /app/interstellar/'
+      - 'install -Dm644 interstellar/data/flutter_assets/assets/icons/logo.png /app/share/icons/hicolor/512x512/apps/one.jwr.interstellar.png'
+      - 'install run.sh /app/bin/interstellar'
+      - 'cp /app/lib/libmpv.so /app/lib/libmpv.so.1'
```

Comment:
why do you need to do that?

Reply:
Sorry, do what?

`cp -r interstellar/ /app/interstellar/`: this copies over the app binaries, libraries, and other assets.
`install -Dm644 interstellar/data/flutter_assets/assets/icons/logo.png /app/share/icons/hicolor/512x512/apps/one.jwr.interstellar.png`: this copies the app icon to the correct location.
`install run.sh /app/bin/interstellar` this copies over the script that runs the app correctly.
`cp /app/lib/libmpv.so /app/lib/libmpv.so.1` libmpv.so is a link to libmpv.so.2 so this makes it so libmpv.so.1 points to the libmpv library, as the the app expects the library to be at libmpv.so.1 instead of libmpv.so.2.

Reply:
> cp /app/lib/libmpv.so /app/lib/libmpv.so.1 libmpv.so is a link to libmpv.so.2 so this makes it so libmpv.so.1 points to the libmpv library, as the the app expects the library to be at libmpv.so.1 instead of libmpv.so.2.

That's the only line I cared.

So basically you are just violating the ABI guarantees instead of building the right version. This is a bad idea. If it works then maybe you are in luck.

Reply:
I certainly hadn't meant to break or violate anything. I'll try to update the .yaml file to build on mpv version 1.

Reply:
The correct libmpv version is now used. Thanks for the help, I wouldn't have seen that otherwise.